src/components/Avatar/Avatar.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import classes from 'classnames';
+import md5 from 'md5';
+import { trim } from 'lodash';
+/**
+ * Displays user profile avatar image with/withour user name
+ */
+const Avatar = ({ className, size, userName, imgUrl, userEmail }) => {
+  const gravatarUrl = userEmail
+    ? `https://www.gravatar.com/avatar/${md5(trim(userEmail.toLowerCase()))}?s=${size}`
+    : imgUrl;
+  const avatarClass = classes('Avatar', className);
+  const avatarSize =
+    size >= 20 && size <= 200 ? { height: size, width: size } : { height: 40, width: 40 };
+  const avatarLetterSize =
+    size >= 20 && size <= 200
+      ? { height: size, width: size, fontSize: size * 0.8 }
+      : { height: 40, width: 40, fontSize: 32 };
+  if (gravatarUrl)
```

Comment:
change up the render logic to 
```js
const name = userName ? userName.charAt(0) : '···';

return (
      <div className={avatarClass} style={avatarSize}>
        {
         imgUrl ||  gravatarUrl ?
          (
            <img
              className="Avatar-img"
              src={imgUrl || gravatarUrl}
              alt={`${userName} avatar image`}
            />
          )
          : 
          name  
        }
      </div>
    );
```
This makes sure that we don't loose our wrapping div when no image is available. The ternary also makes sure that the `imgUrl` is default but is overriden if there's a `userEmail` to generate a gravatar url. If neither `imgUrl` or `userEmail` is present then default to first letter of `userName` or `...`

That's a little confusing to read so best to demonstrate by playing around in storybook